src/test/ui/rfc-2093-infer-outlives/check-dyn-predicates.stderr

```diff
@@ -0,0 +1,21 @@
+error: ["/'a : /'a", "U : /'a"]
+  --> $DIR/check-dyn-predicates.rs:18:1
+   |
+LL | / struct Foo<'a, U> { //~ ERROR 18:1: 21:2: ["/'a : /'a", "U : /'a"]
+LL | |     //~^ ERROR 18:1: 21:2: [Binder(OutlivesPredicate(ReEarlyBound(0, 'a), ReEarlyBound(0, 'a))), Binder(OutlivesPredicate(U, ReEarlyBound(0, 'a)))] [E0640]
+LL | |       value: &'a dyn Iterator<Item = U>
+LL | | }
+   | |_^
+
+error[E0640]: [Binder(OutlivesPredicate(ReEarlyBound(0, 'a), ReEarlyBound(0, 'a))), Binder(OutlivesPredicate(U, ReEarlyBound(0, 'a)))]
```

Comment:
doesn't really matter, but during inference we should probably filter out identity predicates like `'a: 'a`, just for efficiency / to make things easier to read